mlir/lib/Analysis/Presburger/Set.cpp

```diff
@@ -265,36 +267,47 @@ void PresburgerSet::subtract(const PresburgerSet &set) {
 }
 
 bool PresburgerSet::equal(const PresburgerSet &s, const PresburgerSet &t) {
-  // TODO implemenmt this
-  return false;
+  // TODO we cannot assert here, as equal is used by other functionality that
+  // otherwise breaks here
+  // assert(s.getNumSyms() + t.getNumSyms() == 0 &&
+  //       "operations on sets with symbols are not yet supported");
+  if (s.getNumSyms() + t.getNumSyms() != 0)
```

Comment:
Maybe this might be broken, but we didn't fully specify how we want to treat symbols yet. Thus there will be another iteration where we fix all symbol/parameter related stuff (as soon as the PIP stuff has alnded)
.